core/src/test/java/org/elasticsearch/index/store/CorruptedFileIT.java

```diff
@@ -568,34 +590,21 @@ private int numShards(String... index) {
         return shardIterators.size();
     }
 
-    private Map<String, List<Path>> findFilesToCorruptForReplica() throws IOException {
-        Map<String, List<Path>> filesToNodes = new HashMap<>();
-        ClusterState state = client().admin().cluster().prepareState().get().getState();
-        Index test = state.metaData().index("test").getIndex();
-        for (ShardRouting shardRouting : state.getRoutingTable().allShards("test")) {
-            if (shardRouting.primary()) {
```

Comment:
what are your concerns? I think the test is problematic the context of how we require that the new primaries will be selected out of the set of active shards from before the master was restarted - to prevent data loss. I think the old way became brittle and we should fix it anyway?